jobspec2/types.config.go

```diff
@@ -240,6 +242,9 @@ func (c *jobConfig) decodeJob(content *hcl.BodyContent, ctx *hcl.EvalContext) hc
 			continue
 		}
 
+		body := hclutil.BlocksAsAttrs(b.Body)
+		body = dynblock.Expand(body, ctx)
```

Comment:
Kind of an edge case anyways but I want to make sure I understand: does moving this here means we can't have dynamic blocks in the context of a free-standing task or vault block?

Reply:
we can - as the free-standing task/vault blocks are still inside the job block content. This excludes use of dynamic in locals/variables.